### PR TITLE
Accept/reject a recommended item for a gallery

### PR DESCRIPTION
**Pull request by spyrosbek**

### CODE REVIEW

File: src/pages/set/_.vue [hunk 1]
```diff
@@ -165,6 +170,9 @@
     async fetch() {
       try {
         await this.$store.dispatch('set/fetchActive', this.$route.params.pathMatch);
+        if (this.enableRecommendations && this.$auth.loggedIn && this.userIsOwner) {
+          this.$store.dispatch('set/fetchActiveRecommendations', `/${this.$route.params.pathMatch}`);
```
Comment (annachristaki) [inline src/pages/set/_.vue:174]: Make sure this is called on page reload

File: src/store/set.js [hunk 2]
```diff
@@ -166,6 +170,36 @@ export default {
...
+        });
+    },
+    acceptRecommendation({ state, commit }, params) {
+      return this.$apis.recommendation.accept('set', params.setId, params.itemIds)
+        .then(response => {
+          const recList = state.activeRecommendations.slice();
+          const index = recList.map((item) => {
+            return item.id;
+          }).indexOf(params.itemIds[0]);
+          recList.splice(index, 1, response.items[0]);
```
Comment (annachristaki) [inline src/store/set.js:187]: Check that response contains items before getting first one

File: src/store/set.js [hunk 3]
```diff
@@ -166,6 +170,36 @@ export default {
...
+        });
+    },
+    rejectRecommendation({ state, commit }, params) {
+      return this.$apis.recommendation.reject('set', params.setId, params.itemIds)
+        .then(response => {
+          const recList = state.activeRecommendations.slice();
+          const index = recList.map((item) => {
+            return item.id;
+          }).indexOf(params.itemIds[0]);
+          recList.splice(index, 1, response.items[0]);
```
Comment (annachristaki) [inline src/store/set.js:199]: Check that response contains items

File: src/store/set.js [hunk 4]
```diff
@@ -166,6 +170,36 @@ export default {
...
+          commit('setActiveRecommendations', response.items);
+        });
+    },
+    acceptRecommendation({ state, commit }, params) {
+      return this.$apis.recommendation.accept('set', params.setId, params.itemIds)
+        .then(response => {
+          const recList = state.activeRecommendations.slice();
+          const index = recList.map((item) => {
+            return item.id;
+          }).indexOf(params.itemIds[0]);
```
Comment (annachristaki) [inline src/store/set.js:186]: Maybe better to use findIndex

**Review by annachristaki — CHANGES_REQUESTED** [review]
Needs a few changes to work without errors

File: src/components/account/UserButtons.vue [hunk 5]
```diff
@@ -4,20 +4,40 @@
...
-      />
-      <b-button
-        :pressed="liked"
-        class="icon-heart"
-        data-qa="like button"
-        :aria-label="$t('actions.like')"
-        size="sm"
-        @click="toggleLiked"
-      />
+      <template
```
Comment (rwd) [inline src/components/account/UserButtons.vue:7]: Let's move the recommendation buttons and accompanying logic out into a new component, `src/components/recommendation/RecommendationButtons.vue`

Then, in ItemPreviewCard.vue:
```html
    <template v-slot:buttons>
      <RecommendationButtons
        v-if="recommendedItem"
        v-model="identifier"
      />
      <UserButtons
        v-else
        v-model="identifier"
      />
    </template>
```

**Review by rwd — CHANGES_REQUESTED** [review]

**Review by annachristaki — APPROVED** [review]

**Review by lbiedinger — APPROVED** [review]

**Review by rwd — APPROVED** [review]

**Review by lbiedinger — APPROVED** [review]

**Review by lbiedinger — APPROVED** [review]

**Pull request merged**